packages/drivers/odsp-driver/src/getUrlAndHeadersWithAuth.ts

```diff
@@ -32,15 +35,16 @@ export function getUrlAndHeadersWithAuth(url: string, token: string | null): { u
     if (tokenIsQueryParam) {
         // The token itself is a query param
         tokenQueryParam += token.substring(1);
-    } else {
+    } else if (!forceAccessTokenViaAuthorizationHeader) {
         tokenQueryParam += `access_token=${encodeURIComponent(token)}`;
     }
 
     // ODSP APIs have a limitation that the query string cannot exceed 2048 characters.
     // We try to stick the access token in the URL to make it a simple XHR request and avoid an options call.
     // If the query string exceeds 2048, we have to fall back to sending the access token as a header, which
     // has a negative performance implication as it adds a performance overhead.
-    if (tokenIsQueryParam || getQueryStringLength(url + tokenQueryParam) <= 2048) {
+    if (tokenIsQueryParam ||
```

Comment:
Shouldn't forceAccessTokenViaAuthorizationHeader come in AND of the both the previous conditions? since we need to enforce it. Here if tokenIsQueryParam is true and forceAccessTokenViaAuthorizationHeader  is also true then we will still pass it as a query parameter.

Reply:
Changed implementation to remove legacy support for token in shape of query string. Also added unit tests.